k8s/upgrades/0.6.0-0.7.0/oebs_update.sh

```diff
@@ -0,0 +1,186 @@
+
+rollout_status=$(kubectl rollout status --namespace $ns deployment/$r_dep)
+rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+then echo "ERROR: $rc"; exit; fi
+
+#### PATCH TARGET DEPLOYMENT ####
+echo "Upgrading Target Deployment to 0.7"
+kubectl patch deployment  --namespace $ns $c_dep -p "$(cat jiva-target-patch.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+kubectl delete rs $c_rs --namespace $ns
+
+rollout_status=$(kubectl rollout status --namespace $ns  deployment/$c_dep)
+rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+then echo "ERROR: $rc"; exit; fi
+
+#### PATCH TARGET SERVICE ####
+echo "Upgrading Target Service to 0.7"
+kubectl patch service --namespace $ns $c_svc -p "$(cat jiva-target-svc-patch.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+### REMOVE DEPRECATED LABELS
+
+kubectl patch service --namespace $ns $c_svc --type json -p "$(cat target-svc-patch-remove-labels.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+kubectl label svc --namespace $ns $c_svc "vsm-"
+kubectl label svc --namespace $ns $c_svc "openebs/controller-service-"
+
+# Fetch the older target and replica - ReplicaSet objects which need to be 
+# deleted before upgrading. If not deleted, the new pods will be stuck in 
+# creating state - due to affinity rules. 
+c_rs=$(kubectl get rs -o name --namespace $ns | grep $c_dep | cut -d '/' -f 2)
+r_rs=$(kubectl get rs -o name --namespace $ns | grep $r_dep | cut -d '/' -f 2)
+
+
+# PATCH JIVA REPLICA DEPLOYMENT ####
+echo "Remove deprecated lables from Replica Deployment"
+kubectl patch deployment --namespace $ns $r_dep --type json -p "$(cat replica-patch-remove-labels.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+kubectl delete rs $r_rs --namespace $ns
+
+rollout_status=$(kubectl rollout status --namespace $ns deployment/$r_dep)
+rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+then echo "ERROR: $rc"; exit; fi
+
+#### PATCH TARGET DEPLOYMENT ####
+echo "Remove deprecated lables from Controller Deployment"
+kubectl patch deployment  --namespace $ns $c_dep --type json -p "$(cat target-patch-remove-labels.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+kubectl delete rs $c_rs --namespace $ns
+
+rollout_status=$(kubectl rollout status --namespace $ns  deployment/$c_dep)
+rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+then echo "ERROR: $rc"; exit; fi
+
+echo "Clearing temporary files"
+rm jiva-replica-patch.tpl.json.0
+rm jiva-replica-patch.tpl.json.1
+rm jiva-replica-patch.tpl.json.2
+rm jiva-replica-patch.json
+rm jiva-target-patch.tpl.json.0
+rm jiva-target-patch.tpl.json.1
+rm jiva-target-patch.tpl.json.2
+rm jiva-target-patch.json
+rm jiva-target-svc-patch.tpl.json.0
+rm jiva-target-svc-patch.json
+
+echo "Successfully upgraded $pv to 0.7. Please run your application checks."
+exit 0
```

Comment:
Do we also need to take care of updating labels/annotations on PV/PVC objects during upgrade to 0.7 ?

Reply:
good one. had to annotate PV. done.